### PR TITLE
Create OSIAM extension programmatically

### PR DESCRIPTION
**Pull request by tkrille**
Doing it in the database migration is not a good idea. Due to the fact that each migration has a checksum, we would not be able to change what should initially be created. Example data has already been migrated to programmatic creation too.

This also contains a refactoring of the extension DAO. See the corresponding commit for details.

### CODE REVIEW

File: src/test/groovy/org/osiam/scim/extension/OsiamExtensionTest.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.osiam.scim.extension
+
+import org.osiam.resources.scim.ExtensionFieldType
+import org.osiam.storage.ExtensionRepository
+import spock.lang.Specification
+
+class OsiamExtensionTest extends Specification {
```
Comment (dacrome) [inline src/test/groovy/org/osiam/scim/extension/OsiamExtensionTest.groovy]: `OsiamExtensionTest`? I guess `OsiamExtensionSpec`.

Comment (tkrille) [inline src/test/groovy/org/osiam/scim/extension/OsiamExtensionTest.groovy]: Yes, apparently I hit the wrong buttons in IntelliJ :)

**Comment by tkrille** [comment]
Updated

**Comment by dacrome** [comment]
Thank you!

**Pull request merged**